### PR TITLE
made projectile ignore ignored files on windows

### LINKED ISSUE #625: files not ignored by helm-projectile-find-file
I have my .projectile file set up to ignore *.png, *.jpg, *.gif and while they're successfully ignored by `projectile-find-file` I can see them in the `helm-projectile-find-file` results.

### PR DESCRIPTION
**Pull request by mijoharas**
fixes #625 and possibly others (#740 ?).

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -716,7 +716,7 @@ Files are returned as relative paths to the project root."
     ;; cache disabled or cache miss
     (or files-list
         (if (eq projectile-indexing-method 'native)
-            (projectile-dir-files-native root directory)
+            (projectile-remove-ignored (projectile-dir-files-native root directory))
```
Comment (bbatsov) [inline projectile.el]: `projectile-dir-files-native` is doing the ignoring internally. If this is not happening than there's a bug in it.

Comment (mijoharas) [inline projectile.el]: Ah, thanks for pointing that out. I'll find out how it's broken and fix it there.

**Comment by mijoharas** [comment]
I guess I should have read @ahyatt's comment more carefully as he had identified the problem correctly.

**Comment by bbatsov** [comment]
Mention this bugfix in the changelog. You should also mention the related bugs in the commit message (e.g. `[Fix #625] Handle properly ....`).

**Comment by mijoharas** [comment]
Thanks for pointing out that function to me. Should be fixed now.

**Comment by bbatsov** [comment]
@mijoharas Guess you'll have to rebase on top of the current master.

**Comment by mijoharas** [comment]
Sorry, didn't notice the merge conflicts, done now.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**